src/snapred/backend/recipe/algorithm/ReductionAlgorithm.py

```diff
@@ -9,214 +9,129 @@
-        # TODO: Params is supposed to be smallest dmin, smalled dbin, largest dmax
-        # self.enqueueAlgorithm('Rebin', "Rebinning", isChild=False,  InputWorkspace=data,
-        #                       Params='0.338, -0.00086, 5.0', PreserveEvents=False,
-        #                       OutputWorkspace="rebinned_data_before_focus")
-        # data = "rebinned_data_before_focus"
-        # vanadium = self.enqueueAlgorithm('Rebin', "Rebinning", isChild=False, InputWorkspace=vanadium,
-        #                                  Params='0.338, -0.00086, 5.0', PreserveEvents=False,
-        #                                  OutputWorkspace="rebinned_vanadium_before_focus")
-        # vanadium = "rebinned_vanadium_before_focus"
-        # 11 For each Group (no for each loop, the algos apply things based on groups of group workspace)
-        data = self.mantidSnapper.DiffractionFocussing(
-            "Applying Diffraction Focussing...",
-            InputWorkspace=data,
-            GroupingWorkspace=groupingworkspace,
-            OutputWorkspace="focused_data",
-        )
-        vanadium = self.mantidSnapper.DiffractionFocussing(
-            "Applying Diffraction Focussing...",
-            InputWorkspace=vanadium,
-            GroupingWorkspace=groupingworkspace,
-            OutputWorkspace="diffraction_focused_vanadium",
-        )
+    def validateInputs(self) -> Dict[str, str]:
+        return {}
 
-        # 2 NormalizeByCurrent -- just apply to data
-        self.mantidSnapper.NormaliseByCurrent("Normalizing Current ...", InputWorkspace=data, OutputWorkspace=data)
+    def chopIngredients(self, ingredients: ReductionIngredients):
+        self.dMin = ingredients.pixelGroup.dMin()
+        self.dMax = ingredients.pixelGroup.dMax()
+        self.dBin = ingredients.pixelGroup.dBin()
 
-        # self.deleteWorkspace(Workspace=rebinned_data_before_focus)
-        self.mantidSnapper.WashDishes(
-            "Deleting Rebinned Data Before Focus...",
-            Workspace="CommonRed",
+    def unbagGroceries(self):
+        self.outputWorkspace = self.getPropertyValue("OutputWorkspace")
+        self.vanadiumWorkspace = "tmp_vanadium"  # TODO use unique name genetator
+        self.mantidSnapper.CloneWorkspace(
+            "Clone copy of input to serve as output",
+            InputWorkspace=self.getPropertyValue("InputWorkspace"),
+            OutputWorkspace=self.outputWorkspace,
         )
-
-        # compress data
-        # data = self.compressEvents(InputWorkspace=data, OutputWorkspace='event_compressed_data')
-
-        # sum chunks if files are large
-        # TODO: Implement New Strip Peaks that allows for multiple FWHM, one per group,
-        # for now just grab the first one to get it to run
-        peakPositions = ",".join(
-            str(s) for s in reductionIngredients.reductionState.stateConfig.normalizationCalibrant.peaks
+        self.mantidSnapper.CloneWorkspace(
```

Comment:
Are these clones necessary if we are depending on GroceryService to keep our raw data safe?

Reply:
@ekapadi had mentioned before that we should distinguish Inputs from InOuts.  It isn't necessary for the reasons you said, but for the sake of clarity this algorithm doesn't make any changes to its input workspace.  I could rewrite the input workspace as an InOut property

Reply:
I would prefer they be InOuts then

Reply:
Done